app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala

```diff
@@ -124,6 +124,49 @@ class DesConnector @Inject()(httpClient: HttpClient, metrics: Metrics, agentCach
     }
   }
 
+  def getInactiveClientRelationships(taxIdentifier: TaxIdentifier)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[Seq[InactiveRelationship]] = {
+    val encodedClientId = UriEncoding.encodePathSegment(taxIdentifier.value, "UTF-8")
+
+    val url = inactiveClientRelationshipDesUrl(taxIdentifier, encodedClientId)
+
+    getWithDesHeaders("GetInactiveClientRelationships", url).map { response =>
+      response.status match {
+        case Status.OK =>
+          response.json.as[InactiveRelationshipResponse].relationship.filter(isNotActive)
+        case Status.BAD_REQUEST => Seq.empty
+        case Status.NOT_FOUND   => Seq.empty
+        case other: Int =>
+          throw UpstreamErrorResponse(response.body, other, other)
+      }
+    }
+  }
+
+  private def inactiveClientRelationshipDesUrl(taxIdentifier: TaxIdentifier, encodedClientId: String) = {
+    val fromDateString = appConfig.inactiveRelationshipsClientRecordStartDate
+    val from = LocalDate.parse(fromDateString).toString
+    val now = LocalDate.now().toString
+    taxIdentifier match {
+      case MtdItId(_) =>
+        new URL(
```

Comment:
These Strings are very brittle. I would prefer some automatic encoding, so you don't have to repeat `from` and `to` and all the other options several times